packages/design-system/src/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import cx from 'classnames';
+import SearchIcon from 'react-material-icon-svg/dist/MagnifyIcon';
+import CloseIcon from 'react-material-icon-svg/dist/CloseIcon';
+import { debounce } from '@livechat/data-utils';
+
+import styles from './style.scss';
+import { Loader } from '../Loader';
+import { KeyCodes } from '../../constants/keyCodes';
+
+const baseClass = 'search-bar';
+const noop = () => {};
+
+class SearchBar extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.inputRef = React.createRef();
+    this.onChangeDebounced = debounce(this.props.debounceInMs, value => {
+      this.props.onChange(value);
+    });
+
+    this.state = {
+      searchTerm: '',
+      isInCompactMode: this.props.compact
+    };
+  }
+
+  handleChange = e => {
+    const { value } = e.target;
+    const { debounceInMs, onChange } = this.props;
+
+    this.setState({
+      searchTerm: value
+    });
+
+    if (debounceInMs) {
+      this.onChangeDebounced(value);
+    } else {
+      onChange(value);
+    }
+  };
+
+  handleKeyPress = key => {
+    const { onSubmit } = this.props;
+
+    if (onSubmit) {
+      if (key.keyCode === KeyCodes.enter) {
+        onSubmit(this.state.searchTerm);
+      }
+    }
+  };
+
+  handleClear = () => {
+    this.setState({
```

Comment:
you need to inform the parent component about clear if the component is controlled (controlled===has value prop from the outside). Please add `onClear` property (you can't just use onChange :) )

Reply:
Can you be more precise or explain why I can't use onChange / onSubmit with empty value? I just did something like that in this PR. Should i have another prop `onClear`, which will inform user above about clearing input? Right now I don't see it, because I can use onClear() inside SearchBar component - this will delegate to component above and what does it give me?

Reply:
Just add `onClear` to the `searchFunction` definition: `const searchFunction = onClear || onSubmit || onChange;`
If the developer needs different behavior he will use this property. Let's say it won't be a good idea to fetch new data on clear button click - it seems to me like a common case when you need to optimize requests.